phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/engines/EnginesController.java

```diff
@@ -39,78 +40,85 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
 
 import static org.phoenicis.configuration.localisation.Localisation.tr;
 
 public class EnginesController {
     private final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(AppsController.class);
     private final EnginesView enginesView;
-    private final RepositoryManager repositoryManager;
     private final EnginesManager enginesManager;
+
     private ThemeManager themeManager;
-    private RepositoryDTO repositoryCache = null;
+    private RepositoryDTO repositoryCache;
     private Map<String, Engine> enginesCache = new HashMap<>();
     private Map<String, List<EngineSubCategoryDTO>> versionsCache = new HashMap<>();
 
     public EnginesController(EnginesView enginesView, RepositoryManager repositoryManager,
-            EnginesManager enginesManager,
-            ThemeManager themeManager) {
+            EnginesManager enginesManager, ThemeManager themeManager) {
+        super();
+
         this.enginesView = enginesView;
-        this.repositoryManager = repositoryManager;
         this.enginesManager = enginesManager;
         this.themeManager = themeManager;
 
-        this.repositoryManager.addCallbacks(repositoryDTO -> this.enginesManager.fetchAvailableEngines(repositoryDTO,
-                engines -> this.populateView(repositoryDTO, engines),
-                e -> Platform.runLater(() -> enginesView.showFailure(tr("Loading engines failed."), Optional.of(e)))),
+        repositoryManager.addCallbacks(
```

Comment:
Why did your remove `this` in several places?

Reply:
I didn't notice this myself.

The changes reflect my current programming style.
I try to use `this` only for assignments or it's otherwise required, for example in inner classes.
Otherwise I depend on my IDE to show me whether a variable is defined on an object scope or a method scope.

This approach has the benefit in my eyes that it shortens the code.
In addition it makes it easier to see the important object scoped assignments via `this`.

Reply:
I understand. However, the Phoenicis code mainly uses `this` whenever class members are accessed. I think we should keep this style until decided otherwise (especially in this case because the diff shows changes which are not really changes).

Reply:
These are quite a lot of unnecessary `this`.
I think it would be better to change the code specification.
I agree that it is a bad idea to simply change the use of `this` in methods/places you otherwise don't modify.
Therefore I propose to only change it when a method is rewritten.

What do you think?

Reply:
We should discuss this topic in a separate issue and leave the code like it was for this PR.

Reply:
My problem is that I've basically rewritten the whole constructor, `populateView` and added a new method `fetchEngineSubcategories` so it's hard for me revert the code to a state it has never been in.

Reply:
The constructor is very much the same like before. I agree for the methods.